src/hotspot/share/gc/g1/g1CardSetContainers.inline.hpp

```diff
@@ -47,10 +47,14 @@ inline G1CardSetInlinePtr::CardSetPtr G1CardSetInlinePtr::merge(CardSetPtr orig_
 inline G1AddCardResult G1CardSetInlinePtr::add(uint card_idx, uint bits_per_card, uint max_cards_in_inline_ptr) {
   assert(_value_addr != nullptr, "No value address available, cannot add to set.");
 
+  uint cur_idx = 0;
   while (true) {
     uint num_elems = num_cards_in(_value);
+    if (num_elems > 0) {
+      cur_idx = find(card_idx, bits_per_card, cur_idx, num_elems);
```

Comment:
I found the naming here a bit confusing, with `num_elems` here but `end_at` in the find function.  Similarly in `contains`.

Reply:
Fixed, opted to use `num_elems` in find function too.